src/mmap_unix.rs

```diff
@@ -52,53 +74,75 @@ impl MmapRegion {
     ///
     /// # Arguments
     /// * `size` - Size of memory region in bytes.
-    pub fn new(size: usize) -> io::Result<Self> {
-        // This is safe because we are creating an anonymous mapping in a place not already used by
-        // any other area in this process.
-        let addr = unsafe {
-            libc::mmap(
-                null_mut(),
-                size,
-                libc::PROT_READ | libc::PROT_WRITE,
-                libc::MAP_ANONYMOUS | libc::MAP_SHARED | libc::MAP_NORESERVE,
-                -1,
-                0,
-            )
-        };
-        if addr == libc::MAP_FAILED {
-            return Err(io::Error::last_os_error());
-        }
-        Ok(Self {
-            addr: addr as *mut u8,
+    pub fn new(size: usize) -> Result<Self> {
+        Self::build(
+            None,
             size,
-        })
+            libc::PROT_READ | libc::PROT_WRITE,
+            libc::MAP_ANONYMOUS | libc::MAP_NORESERVE | libc::MAP_PRIVATE,
+        )
     }
 
     /// Maps the `size` bytes starting at `offset` bytes of the given `fd`.
     ///
     /// # Arguments
-    /// * `fd` - File descriptor to mmap from.
+    /// * `file_offset` - File object and offset to mmap from.
     /// * `size` - Size of memory region in bytes.
-    /// * `offset` - Offset in bytes from the beginning of `fd` to start the mmap.
-    pub fn from_fd(fd: &AsRawFd, size: usize, offset: libc::off_t) -> io::Result<Self> {
-        // This is safe because we are creating a mapping in a place not already used by any other
-        // area in this process.
-        let addr = unsafe {
-            libc::mmap(
-                null_mut(),
-                size,
-                libc::PROT_READ | libc::PROT_WRITE,
-                libc::MAP_SHARED,
-                fd.as_raw_fd(),
-                offset,
-            )
+    pub fn from_file(file_offset: FileOffset, size: usize) -> Result<Self> {
```

Comment:
We need to modify the same way the function `from_fd` in `mmap_windows.rs`. This should be pretty straightforward.